src/repositories/indexable-repository.php

```diff
@@ -314,100 +262,31 @@ public function find_by_multiple_ids_and_type( $object_ids, $object_type, $auto_
 	}
 
 	/**
-	 * Creates an indexable by its ID and type.
+	 * Returns all ancestors of a given indexable. Optionally prepending a set of static ancestors.
 	 *
-	 * @param int    $object_id   The indexable object ID.
-	 * @param string $object_type The indexable object type.
+	 * @param Indexable $indexable              The indexable to find the ancestors of.
+	 * @param array     $static_ancestor_wheres Optional. The where conditions by which to find static ancestors.
+	 *                                          Should be an array of arrays with the inner arrays containing find
+	 *                                          conditions.
 	 *
-	 * @return bool|Indexable Instance of indexable.
+	 * @return Indexable[] All ancestors of the given indexable.
 	 */
-	public function create_for_id_and_type( $object_id, $object_type ) {
-		/**
-		 * Indexable instance.
-		 *
-		 * @var Indexable $indexable
-		 */
-		$indexable = $this->query()->create();
-
-		switch ( $object_type ) {
-			case 'post':
-				$indexable = $this->post_builder->build( $object_id, $indexable );
-				break;
-			case 'user':
-				$indexable = $this->author_builder->build( $object_id, $indexable );
-				break;
-			case 'term':
-				$indexable = $this->term_builder->build( $object_id, $indexable );
-				break;
+	public function get_ancestors( Indexable $indexable, $static_ancestor_wheres = [] ) {
+		$ancestor_query = $this->query()
+							   ->table_alias( 'i' )
+							   ->select( 'i.*' )
+							   ->join( 'indexable_hierarchy', 'i.id = ih.ancestor_id', 'ih' )
+							   ->where( 'ih.indexable_id', $indexable->id )
+							   ->order_by_desc( 'ih.depth' );
```

Comment:
Weird alignment

Reply:
That's because GitHub is always configured to preview files with a tab being 8 spaces instead of 4. In an editor this would appear as expected.